File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/UploadAndExtractZipDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Details to create or update a workspace zip upload and extraction operation.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=UploadAndExtractZipDetails.Builder.class)

public final class UploadAndExtractZipDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"action", "operationKey", "zipFileName", "destinationFolderPath", "isOverwrite", "eTag", "sizeInBytes"})
    public UploadAndExtractZipDetails(Action action, String operationKey, String zipFileName, String destinationFolderPath, Boolean isOverwrite, String eTag, Long sizeInBytes) {
        super();
        this.action = action;
        this.operationKey = operationKey;
        this.zipFileName = zipFileName;
        this.destinationFolderPath = destinationFolderPath;
        this.isOverwrite = isOverwrite;
        this.eTag = eTag;
        this.sizeInBytes = sizeInBytes;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Action to perform for the upload and extract operation. CREATE starts the async operation and returns an upload PAR URL. UPDATE records final uploaded object metadata for an existing operation.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("action")
private Action action;

        /**
         * Action to perform for the upload and extract operation. CREATE starts the async operation and returns an upload PAR URL. UPDATE records final uploaded object metadata for an existing operation.
         * @param action the value to set
         * @return this builder
         **/
        

public Builder action(Action action) {
    this.action = action;
    return this;
}
            /**
     * Async operation key returned by the CREATE action. Required for UPDATE.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("operationKey")
private String operationKey;

        /**
         * Async operation key returned by the CREATE action. Required for UPDATE.
         * @param operationKey the value to set
         * @return this builder
         **/
        

public Builder operationKey(String operationKey) {
    this.operationKey = operationKey;
    return this;
}
            /**
     * Name of the zip file that will be uploaded through the returned PAR URL. Required for CREATE.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("zipFileName")
private String zipFileName;

        /**
         * Name of the zip file that will be uploaded through the returned PAR URL. Required for CREATE.
         * @param zipFileName the value to set
         * @return this builder
         **/
        

public Builder zipFileName(String zipFileName) {
    this.zipFileName = zipFileName;
    return this;
}
            /**
     * Workspace folder where the zip root folder will be extracted. Required for CREATE.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("destinationFolderPath")
private String destinationFolderPath;

        /**
         * Workspace folder where the zip root folder will be extracted. Required for CREATE.
         * @param destinationFolderPath the value to set
         * @return this builder
         **/
        

public Builder destinationFolderPath(String destinationFolderPath) {
    this.destinationFolderPath = destinationFolderPath;
    return this;
}
            /**
     * Flag to overwrite existing files during extraction. If false, existing files are skipped.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("isOverwrite")
private Boolean isOverwrite;

        /**
         * Flag to overwrite existing files during extraction. If false, existing files are skipped.
         * @param isOverwrite the value to set
         * @return this builder
         **/
        

public Builder isOverwrite(Boolean isOverwrite) {
    this.isOverwrite = isOverwrite;
    return this;
}
            /**
     * Final Object Storage ETag for the uploaded zip object. If multipart upload is used, the caller commits the multipart upload directly with Object Storage before calling UPDATE.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("eTag")
private String eTag;

        /**
         * Final Object Storage ETag for the uploaded zip object. If multipart upload is used, the caller commits the multipart upload directly with Object Storage before calling UPDATE.
         * @param eTag the value to set
         * @return this builder
         **/
        

public Builder eTag(String eTag) {
    this.eTag = eTag;
    return this;
}
            /**
     * Final uploaded zip object size in bytes. Required for UPDATE.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("sizeInBytes")
private Long sizeInBytes;

        /**
         * Final uploaded zip object size in bytes. Required for UPDATE.
         * @param sizeInBytes the value to set
         * @return this builder
         **/
        

public Builder sizeInBytes(Long sizeInBytes) {
    this.sizeInBytes = sizeInBytes;
    return this;
}


        public UploadAndExtractZipDetails build() {
            UploadAndExtractZipDetails model = new UploadAndExtractZipDetails(this.action
                , this.operationKey
                , this.zipFileName
                , this.destinationFolderPath
                , this.isOverwrite
                , this.eTag
                , this.sizeInBytes);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(UploadAndExtractZipDetails model) {
                this.action(model.getAction());
    this.operationKey(model.getOperationKey());
    this.zipFileName(model.getZipFileName());
    this.destinationFolderPath(model.getDestinationFolderPath());
    this.isOverwrite(model.getIsOverwrite());
    this.eTag(model.getETag());
    this.sizeInBytes(model.getSizeInBytes());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    

    /**
     * Action to perform for the upload and extract operation. CREATE starts the async operation and returns an upload PAR URL. UPDATE records final uploaded object metadata for an existing operation.
     **/
    public enum Action implements com.oracle.bmc.http.internal.BmcEnum {
        Create("CREATE"),
        Update("UPDATE"),
        ;

        

        private final String value;
        private static java.util.Map<String, Action> map;

        static {
            map = new java.util.HashMap<>();
            for (Action v : Action.values()) {
                    map.put(v.getValue(), v);
                
            }
        }

        Action(String value) {
            this.value = value;
        }

        @com.fasterxml.jackson.annotation.JsonValue
        public String getValue() {
            return value;
        }

        @com.fasterxml.jackson.annotation.JsonCreator
        public static Action create(String key) {
            if (map.containsKey(key)) {
                return map.get(key);
            }
            throw new IllegalArgumentException("Invalid Action: " + key);
        }
    };
        /**
     * Action to perform for the upload and extract operation. CREATE starts the async operation and returns an upload PAR URL. UPDATE records final uploaded object metadata for an existing operation.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("action")
    private final Action action;

        /**
     * Action to perform for the upload and extract operation. CREATE starts the async operation and returns an upload PAR URL. UPDATE records final uploaded object metadata for an existing operation.
     * @return the value
     **/
    
    public Action getAction() {
        return action;
    }


        /**
     * Async operation key returned by the CREATE action. Required for UPDATE.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("operationKey")
    private final String operationKey;

        /**
     * Async operation key returned by the CREATE action. Required for UPDATE.
     * @return the value
     **/
    
    public String getOperationKey() {
        return operationKey;
    }


        /**
     * Name of the zip file that will be uploaded through the returned PAR URL. Required for CREATE.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("zipFileName")
    private final String zipFileName;

        /**
     * Name of the zip file that will be uploaded through the returned PAR URL. Required for CREATE.
     * @return the value
     **/
    
    public String getZipFileName() {
        return zipFileName;
    }


        /**
     * Workspace folder where the zip root folder will be extracted. Required for CREATE.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("destinationFolderPath")
    private final String destinationFolderPath;

        /**
     * Workspace folder where the zip root folder will be extracted. Required for CREATE.
     * @return the value
     **/
    
    public String getDestinationFolderPath() {
        return destinationFolderPath;
    }


        /**
     * Flag to overwrite existing files during extraction. If false, existing files are skipped.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("isOverwrite")
    private final Boolean isOverwrite;

        /**
     * Flag to overwrite existing files during extraction. If false, existing files are skipped.
     * @return the value
     **/
    
    public Boolean getIsOverwrite() {
        return isOverwrite;
    }


        /**
     * Final Object Storage ETag for the uploaded zip object. If multipart upload is used, the caller commits the multipart upload directly with Object Storage before calling UPDATE.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("eTag")
    private final String eTag;

        /**
     * Final Object Storage ETag for the uploaded zip object. If multipart upload is used, the caller commits the multipart upload directly with Object Storage before calling UPDATE.
     * @return the value
     **/
    
    public String getETag() {
        return eTag;
    }


        /**
     * Final uploaded zip object size in bytes. Required for UPDATE.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("sizeInBytes")
    private final Long sizeInBytes;

        /**
     * Final uploaded zip object size in bytes. Required for UPDATE.
     * @return the value
     **/
    
    public Long getSizeInBytes() {
        return sizeInBytes;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("UploadAndExtractZipDetails(");
        sb.append("action=").append(String.valueOf(this.action));
        sb.append(", operationKey=").append(String.valueOf(this.operationKey));
        sb.append(", zipFileName=").append(String.valueOf(this.zipFileName));
        sb.append(", destinationFolderPath=").append(String.valueOf(this.destinationFolderPath));
        sb.append(", isOverwrite=").append(String.valueOf(this.isOverwrite));
        sb.append(", eTag=").append(String.valueOf(this.eTag));
        sb.append(", sizeInBytes=").append(String.valueOf(this.sizeInBytes));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof UploadAndExtractZipDetails)) {
            return false;
        }

        UploadAndExtractZipDetails other = (UploadAndExtractZipDetails) o;
        return java.util.Objects.equals(this.action, other.action) &&
            java.util.Objects.equals(this.operationKey, other.operationKey) &&
            java.util.Objects.equals(this.zipFileName, other.zipFileName) &&
            java.util.Objects.equals(this.destinationFolderPath, other.destinationFolderPath) &&
            java.util.Objects.equals(this.isOverwrite, other.isOverwrite) &&
            java.util.Objects.equals(this.eTag, other.eTag) &&
            java.util.Objects.equals(this.sizeInBytes, other.sizeInBytes);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.action == null ? 43 : this.action.hashCode());
        result = (result * PRIME) + (this.operationKey == null ? 43 : this.operationKey.hashCode());
        result = (result * PRIME) + (this.zipFileName == null ? 43 : this.zipFileName.hashCode());
        result = (result * PRIME) + (this.destinationFolderPath == null ? 43 : this.destinationFolderPath.hashCode());
        result = (result * PRIME) + (this.isOverwrite == null ? 43 : this.isOverwrite.hashCode());
        result = (result * PRIME) + (this.eTag == null ? 43 : this.eTag.hashCode());
        result = (result * PRIME) + (this.sizeInBytes == null ? 43 : this.sizeInBytes.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/upload_and_extract_zip_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class UploadAndExtractZipDetails(object):
    """
    Details to create or update a workspace zip upload and extraction operation.
    """

    #: A constant which can be used with the action property of a UploadAndExtractZipDetails.
    #: This constant has a value of "CREATE"
    ACTION_CREATE = "CREATE"

    #: A constant which can be used with the action property of a UploadAndExtractZipDetails.
    #: This constant has a value of "UPDATE"
    ACTION_UPDATE = "UPDATE"

    def __init__(self, **kwargs):
        """
        Initializes a new UploadAndExtractZipDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param action:
            The value to assign to the action property of this UploadAndExtractZipDetails.
            Allowed values for this property are: "CREATE", "UPDATE"
        :type action: str

        :param operation_key:
            The value to assign to the operation_key property of this UploadAndExtractZipDetails.
        :type operation_key: str

        :param zip_file_name:
            The value to assign to the zip_file_name property of this UploadAndExtractZipDetails.
        :type zip_file_name: str

        :param destination_folder_path:
            The value to assign to the destination_folder_path property of this UploadAndExtractZipDetails.
        :type destination_folder_path: str

        :param is_overwrite:
            The value to assign to the is_overwrite property of this UploadAndExtractZipDetails.
        :type is_overwrite: bool

        :param e_tag:
            The value to assign to the e_tag property of this UploadAndExtractZipDetails.
        :type e_tag: str

        :param size_in_bytes:
            The value to assign to the size_in_bytes property of this UploadAndExtractZipDetails.
        :type size_in_bytes: int

        """
        self.swagger_types = {
            'action': 'str',
            'operation_key': 'str',
            'zip_file_name': 'str',
            'destination_folder_path': 'str',
            'is_overwrite': 'bool',
            'e_tag': 'str',
            'size_in_bytes': 'int'
        }

        self.attribute_map = {
            'action': 'action',
            'operation_key': 'operationKey',
            'zip_file_name': 'zipFileName',
            'destination_folder_path': 'destinationFolderPath',
            'is_overwrite': 'isOverwrite',
            'e_tag': 'eTag',
            'size_in_bytes': 'sizeInBytes'
        }

        self._action = None
        self._operation_key = None
        self._zip_file_name = None
        self._destination_folder_path = None
        self._is_overwrite = None
        self._e_tag = None
        self._size_in_bytes = None

    @property
    def action(self):
        """
        **[Required]** Gets the action of this UploadAndExtractZipDetails.
        Action to perform for the upload and extract operation. CREATE starts the async operation and returns an upload PAR URL. UPDATE records final uploaded object metadata for an existing operation.

        Allowed values for this property are: "CREATE", "UPDATE"


        :return: The action of this UploadAndExtractZipDetails.
        :rtype: str
        """
        return self._action

    @action.setter
    def action(self, action):
        """
        Sets the action of this UploadAndExtractZipDetails.
        Action to perform for the upload and extract operation. CREATE starts the async operation and returns an upload PAR URL. UPDATE records final uploaded object metadata for an existing operation.


        :param action: The action of this UploadAndExtractZipDetails.
        :type: str
        """
        allowed_values = ["CREATE", "UPDATE"]
        if not value_allowed_none_or_none_sentinel(action, allowed_values):
            raise ValueError(
                "Invalid value for `action`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._action = action

    @property
    def operation_key(self):
        """
        Gets the operation_key of this UploadAndExtractZipDetails.
        Async operation key returned by the CREATE action. Required for UPDATE.


        :return: The operation_key of this UploadAndExtractZipDetails.
        :rtype: str
        """
        return self._operation_key

    @operation_key.setter
    def operation_key(self, operation_key):
        """
        Sets the operation_key of this UploadAndExtractZipDetails.
        Async operation key returned by the CREATE action. Required for UPDATE.


        :param operation_key: The operation_key of this UploadAndExtractZipDetails.
        :type: str
        """
        self._operation_key = operation_key

    @property
    def zip_file_name(self):
        """
        Gets the zip_file_name of this UploadAndExtractZipDetails.
        Name of the zip file that will be uploaded through the returned PAR URL. Required for CREATE.


        :return: The zip_file_name of this UploadAndExtractZipDetails.
        :rtype: str
        """
        return self._zip_file_name

    @zip_file_name.setter
    def zip_file_name(self, zip_file_name):
        """
        Sets the zip_file_name of this UploadAndExtractZipDetails.
        Name of the zip file that will be uploaded through the returned PAR URL. Required for CREATE.


        :param zip_file_name: The zip_file_name of this UploadAndExtractZipDetails.
        :type: str
        """
        self._zip_file_name = zip_file_name

    @property
    def destination_folder_path(self):
        """
        Gets the destination_folder_path of this UploadAndExtractZipDetails.
        Workspace folder where the zip root folder will be extracted. Required for CREATE.


        :return: The destination_folder_path of this UploadAndExtractZipDetails.
        :rtype: str
        """
        return self._destination_folder_path

    @destination_folder_path.setter
    def destination_folder_path(self, destination_folder_path):
        """
        Sets the destination_folder_path of this UploadAndExtractZipDetails.
        Workspace folder where the zip root folder will be extracted. Required for CREATE.


        :param destination_folder_path: The destination_folder_path of this UploadAndExtractZipDetails.
        :type: str
        """
        self._destination_folder_path = destination_folder_path

    @property
    def is_overwrite(self):
        """
        Gets the is_overwrite of this UploadAndExtractZipDetails.
        Flag to overwrite existing files during extraction. If false, existing files are skipped.


        :return: The is_overwrite of this UploadAndExtractZipDetails.
        :rtype: bool
        """
        return self._is_overwrite

    @is_overwrite.setter
    def is_overwrite(self, is_overwrite):
        """
        Sets the is_overwrite of this UploadAndExtractZipDetails.
        Flag to overwrite existing files during extraction. If false, existing files are skipped.


        :param is_overwrite: The is_overwrite of this UploadAndExtractZipDetails.
        :type: bool
        """
        self._is_overwrite = is_overwrite

    @property
    def e_tag(self):
        """
        Gets the e_tag of this UploadAndExtractZipDetails.
        Final Object Storage ETag for the uploaded zip object. If multipart upload is used, the caller commits the multipart upload directly with Object Storage before calling UPDATE.


        :return: The e_tag of this UploadAndExtractZipDetails.
        :rtype: str
        """
        return self._e_tag

    @e_tag.setter
    def e_tag(self, e_tag):
        """
        Sets the e_tag of this UploadAndExtractZipDetails.
        Final Object Storage ETag for the uploaded zip object. If multipart upload is used, the caller commits the multipart upload directly with Object Storage before calling UPDATE.


        :param e_tag: The e_tag of this UploadAndExtractZipDetails.
        :type: str
        """
        self._e_tag = e_tag

    @property
    def size_in_bytes(self):
        """
        Gets the size_in_bytes of this UploadAndExtractZipDetails.
        Final uploaded zip object size in bytes. Required for UPDATE.


        :return: The size_in_bytes of this UploadAndExtractZipDetails.
        :rtype: int
        """
        return self._size_in_bytes

    @size_in_bytes.setter
    def size_in_bytes(self, size_in_bytes):
        """
        Sets the size_in_bytes of this UploadAndExtractZipDetails.
        Final uploaded zip object size in bytes. Required for UPDATE.


        :param size_in_bytes: The size_in_bytes of this UploadAndExtractZipDetails.
        :type: int
        """
        self._size_in_bytes = size_in_bytes

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
